codegen/src/types/mod.rs

```diff
@@ -83,7 +84,7 @@ impl<'a> TypeGenerator<'a> {
     }
 
     /// Generate a module containing all types defined in the supplied type registry.
-    pub fn generate_types_mod(&self) -> Module {
+    pub fn generate_types_mod(&self, crate_path: &CratePath) -> Module {
```

Comment:
Perhaps the `crate_path` is better as a property to avoid having to pass around?